framework/views/lazyRepeat.js

```diff
@@ -0,0 +1,282 @@
+        }
+        else if (this._delegate.destroyItemContent) {
+          this._delegate.destroyItemContent(i, element.element.children()[0]);
+        }
+
+        element.element.remove();
+        element.scope.$destroy();
+
+        delete this._renderedElements[i];
+      },
+
+      _calculateStartIndex: function(current) {
+        var start = 0,
+          end = this._maxIndex;
+
+        // Binary search for index at top of screen so
+        // we can speed up rendering.
+        while (true) {
+          var middle = Math.floor((start + end) / 2),
+            value = current + this._itemHeightSum[middle];
+
+          if (end < start) {
+            return 0;
+          }
+          else if (value >= 0 && value - this._getItemHeight(middle) < 0) {
+            return middle;
+          }
+          else if (isNaN(value) || value >= 0) {
+            end = middle - 1;
+          }
+          else {
+            start = middle + 1;
+          }
+
+        }
+      },
+
+      _getItemsInView: function() {
+        var topOffset = this._getTopOffset(),
+          topPosition = topOffset,
+          cnt = this._countItems();
+
+        var startIndex = this._calculateStartIndex(topPosition);
+        startIndex = Math.max(startIndex - 30, 0);
+
+        if (startIndex > 0) {
+          topPosition += this._itemHeightSum[startIndex - 1];
+        }
+
+        var items = [];
+        for (var i = startIndex; i < cnt && topPosition < 4 * window.innerHeight; i++) {
+          var h = this._getItemHeight();
+
+          if (i >= this._itemHeightSum.length) {
+            this._itemHeightSum = this._itemHeightSum.concat(new Array(100));
+          }
+
+          if (i > 0) {
+            this._itemHeightSum[i] = this._itemHeightSum[i - 1] + h;
+          }
+          else {
+            this._itemHeightSum[i] = h;
+          }
+
+          this._maxIndex = Math.max(i, this._maxIndex);
+
+          items.push({
+            index: i,
+            top: topPosition - topOffset
+          });
+
+          topPosition += h;
+        }
+
+        return items;
+      },
+
+      _addSpecialProperties: function(i, scope) {
+        scope.$index = i;
+        scope.$first = i === 0;
+        scope.$last = i === this._countItems() - 1;
+        scope.$middle = !scope.$first && !scope.$last;
+        scope.$even = i % 2 === 0;
+        scope.$odd = !scope.$even;
+      },
+
+      _onScroll: function() {
+        this._render();
+      },
+
+      _addEventListeners: function() {
+        this._bindedOnScroll = this._onScroll.bind(this); 
+        $document[0].addEventListener('scroll', this._bindedOnScroll, true);
+      },
+
+      _removeEventListeners: function() {
+        $document[0].removeEventListener('scroll', this._bindedOnScroll, true);
+      },
+      
+      _destroy: function() {
+        this._removeEventListeners(); 
+        this._element = this._scope = this._attrs = null;
```

Comment:
Request: It is seem that "this._parentElement" and "this._renderedElements" is not released. Could you confirm that it is no DOM leaks?